### PR TITLE
Upgrade meilisearch packages

### PR DESCRIPTION
**Pull request by mdubus**
# Pull Request

## What does this PR do?
- Bump `@meilisearch/instant-meilisearch` to v1.0.0 & update code accordingly (addition of `.searchClient` to the object returned from the core function as specified [on the release notes](https://github.com/meilisearch/meilisearch-js-plugins/releases/tag/%40meilisearch%2Finstant-meilisearch%401.0.0))
- Bump `meilisearch` to v0.36.0
- Bump `react-instantsearch-dom` to v6.40.4
- Update Meilisearch version in the CI

## PR checklist
Please check if your PR fulfills the following requirements:
- [ ] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
       options: --user 1001
     services:
       meilisearch:
-        image: getmeili/meilisearch:v1.0.2
+        image: getmeili/meilisearch:v1.5
```
Comment (curquiza) [inline .github/workflows/tests.yml:64]: @brunoocasali @mdubus shouldn't we use `latest` instead like in all the integration? We could avoid this kind of manual update

Comment (curquiza) [reply .github/workflows/tests.yml:64]: Also the risk currently we had/have here is the test are green for v1.0.2 but not for v1.6.0 when out and we wouldn't even notice it.

Comment (mdubus) [reply .github/workflows/tests.yml:64]: Thanks for your insight @curquiza 🙏 
What we usually did before was update the image with the pre-release Meilisearch version, that way we were almost sure that there won't be no problem. 
I'm pretty sure that's why we never put `latest` in the CI for the meilisearch version. 
Would that work for you if I update the `v1.6.0-rc.0` tag?

Comment (mdubus) [reply .github/workflows/tests.yml]: Gotcha 👍  Thanks for taking the time to explain this to me. 
I updated the CI file to use the `latest` instead :)

**Review by curquiza — APPROVED** [review]

**Comment by mdubus** [comment]
bors merge

**Pull request merged**